### PR TITLE
Softmax op optimization for inference 

### PR DESCRIPTION
**Pull request by jczaja**
This PR is removing Noise Adding for Softmax forward op during inference. Noise adding was needed in training when there was softmax op followed by cross entropy loss. 

Capi DAM test's profiling shows **~5%** improvement in softmax op execution with this optimization.
**Num threads:** 1
**Batch:**             1,8,32,128
**Platform:**         Intel(R) Xeon(R) Platinum 8180 CPU @ 2.50GHz

### CODE REVIEW

**Comment by jczaja** [comment]
@luotao1 I'm experiencing problems with Windows build. Could you advice?

```
[15:29:01]Failed to perform checkout on agent: '"C:\Program Files\Git\bin\git.exe" clean -f -d -x' command failed.
exit code: 1
stderr: warning: failed to remove build/: Permission denied'
```

**Comment by luotao1** [comment]
You can ignore the windows CI now, since it's not `Required` one.

**Comment by luotao1** [comment]
I test on E5-2620 v3. After this PR, the elapsed time of Softmax Op (test on 1000 samples) **from 35630.7ms to 32446.3ms**, about 8.9% speedup on Softmax Op.

File: paddle/fluid/operators/math/softmax_impl.h [hunk 1]
```diff
@@ -65,6 +65,39 @@ void SoftmaxFunctor<DeviceContext, T>::operator()(const DeviceContext& context,
...
+
+    Eigen::DSizes<int, 1> along_class(kClassDim);
+    Eigen::DSizes<int, 2> batch_by_one(batch_size, 1);
+    Eigen::DSizes<int, 2> one_by_class(1, num_classes);
+
+    auto shifted_logits = (logits -
+                           logits.maximum(along_class)
+                               .eval()
+                               .reshape(batch_by_one)
+                               .broadcast(one_by_class));
```
Comment (luotao1) [inline paddle/fluid/operators/math/softmax_impl.h:89]: The only difference between `class SoftmaxFunctor<DeviceContext, T, is_test>` and `class SoftmaxFunctor<DeviceContext, T, true>` is line 57: `unaryExpr(ValueClip<T>())`.

Thus, could you combine `class SoftmaxFunctor<DeviceContext, T, true>` into `class SoftmaxFunctor<DeviceContext, T, is_test`, to make the coder much cleaner and less?

Comment (jczaja) [reply paddle/fluid/operators/math/softmax_impl.h:89]: @luotao1 You are right, but I 'm working on further changes (MKL replacing Eigen + omp vectorization) that will change this code much more

Comment (luotao1) [reply paddle/fluid/operators/math/softmax_impl.h:89]: Got it

Comment (jczaja) [reply paddle/fluid/operators/math/softmax_impl.h:89]: To make it a bit clearer, I'm planning to add MKL+vectorization in Next PR. So after review is done I would like this one merged.

**Review by luotao1 — APPROVED** [review]
LGTM!

File: paddle/fluid/operators/math/softmax_impl.h (hunk 1 above)
Comment (luotao1) [reply paddle/fluid/operators/math/softmax_impl.h:89]: Hope the Next PR! 
By the way, when will you enhance https://github.com/PaddlePaddle/Paddle/pull/14053#discussion_r229623740?

Comment (jczaja) [reply paddle/fluid/operators/math/softmax_impl.h:89]: @luotao1 Sorry for delay with Max sequence pool followup PR. Once I'm done with softmax I will return to #14053 to fix/test suggesstions.
